core/src/main/java/jenkins/model/ModelObjectWithContextMenu.java

```diff
@@ -248,6 +250,18 @@ public void run(JellyContext context, XMLOutput output) throws JellyTagException
         public String icon;
 
         /**
+         * Icon class specification.
+         */
+        @Exported
+        public String iconClassSpec;
```

Comment:
I'm bit confused about the interaction between this field and the `icon` field, which also defines `Icon.classSpec`. It seems like either we should have `iconDef` as a field and remove `iconClassSpec` and `useCSSIconRendering`, or we keep the latter and remove `iconDef`.

Without knowing too much, it feels like a typed `iconDef` field is preferrable, and you can mark `iconDef` with `@Exported` to expose its properties with `MenuItem`.

I'm afraid I don't fully understand how the whole pluggable icon system is designed (and this would be a good opportunity for me to interview you to learn that and I'd be happy to write it down), but I'd think there'd be some constant that abstractly describe the icon (e.g., "this is a right arrow icon" without a specific URL or size), then someone else is responsible for resolving that to a specific URL or a fragment of HTML?